dvc/dependency/__init__.py

```diff
@@ -42,36 +47,47 @@
 # cached, see -o and -O flags for `dvc run`) and 'metric' (whether or not
 # output is a metric file and how to parse it, see `-M` flag for `dvc run`).
 SCHEMA = output.SCHEMA.copy()
-del SCHEMA[schema.Optional(OutputBase.PARAM_CACHE)]
-del SCHEMA[schema.Optional(OutputBase.PARAM_METRIC)]
+del SCHEMA[schema.Optional(OutputBASE.PARAM_CACHE)]
+del SCHEMA[schema.Optional(OutputBASE.PARAM_METRIC)]
+SCHEMA[schema.Optional(DependencyLOCAL.PARAM_PKG)] = schema.Or(
+    str, Config.SECTION_PKG_SCHEMA
+)
 
 
-def _get(stage, p, info):
+def Dependency(stage, info):
```

Comment:
very confusing - functions should be lowercase. I though for a moment that this is a class name

Reply:
can we actually move it as a static method to the DependencyBASE ... and rename DependencyBASE to a Dependency?

Reply:
> very confusing - functions should be lowercase. I though for a moment that this is a class name

This is just a convenience wrapper that returns needed class instance based on arguments. We have similar ones in remote and pkg.

> can we actually move it as a static method to the DependencyBASE ... and rename DependencyBASE to a Dependency?

I don't think that would be suitable, since that method would then have to import all remote classes, which would unnecessarily create an import cycle that would need to be resolved by dynamically importing them.

Reply:
Yep, I understand the problem of the cyclic dependencies. It still not PEP8 and confusing to see a factory _method_ used as a class. Either there should be a pythonic way of doing factories, or let's rename it at least - even `create_dependency` would be less confusing.

Reply:
Ok, moving to OOP Factories.